grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy

```diff
@@ -189,6 +190,10 @@ class Invoice implements Serializable {
         return invoiceType?.code == InvoiceTypeCode.PREPAYMENT_INVOICE
     }
 
+    boolean getIsRegularInvoice() {
+        return invoiceType?.code == InvoiceTypeCode.INVOICE
+    }
+
```

Comment:
I know you've just followed the convention, but I think this would be a nice tech debt topic for the future to discuss whether boolean names should start with the `is` prefix, or it should be reserved for the getter.
Groovy is specific in this case, because we'll access the field without having to use `get` prefix, but I know in Java the "best practices" are NOT to use `is` for boolean field name and use it for the getter.

This is not the topic to discuss under this PR, but just wanted to point it out maybe for the future.

Reply:
> I know you've just followed the convention, but I think this would be a nice tech debt topic for the future to discuss whether boolean names should start with the is prefix

I agree we should have a tech huddle about this to enumerate the conventions and document the one we want to standardize on. 

I always forget what the correct convention for booleans is, but I prefer the Groovy/Grails convention of invoking the getter method through a property access so I think I like the way @awalkowiak did it above. 
```
Boolean getIsApprovalRequired() { 
    
}
if (requisition.origin.isApprovalRequired) {
    ...
}
``` 
vs 
``` 
Boolean IsApprovalRequired() { 
    ...
}
if (requisition.isApprovalRequired()) {
    ...
}
```

> but I know in Java the "best practices" are NOT to use is for boolean field name and use it for the getter.

Unless I'm misunderstanding, Java does provide a design pattern for boolean properties that uses "is".
![image](https://github.com/openboxes/openboxes/assets/1162935/8fa6244d-9890-43b2-a846-809bccadfb76)

The slight difference is that Java is allowing for a is<Property>() method while get getIs<Property>() method allows us to use the Groovy getter convention.

Reply:
> Unless I'm misunderstanding, Java does provide a design pattern for boolean properties that uses "is".

Yes, but they use it for the getter, not the for the property itself. Even the jackson tool when serializing an object would return a boolean without the `is`.
Assuming you'd have a property:
```java
private boolean isValid;
```
and if we returned this object from the API, it'd return it like:
```json
{
  "valid": true,
}
```
The problem with naming the boolean using the `is` prefix is that naming the getter as `getIs` feels weird.

Once again - in Groovy we don't have that much problem with that, because don't have to use getters, but to access the properties directly, so we don't have to do: `something.getIsValid()`, but `something.isValid`, so it's fine.

https://stackoverflow.com/questions/5322648/for-a-boolean-field-what-is-the-naming-convention-for-its-getter-setter